### DISCUSSION TITLE
allow standalone usage

### DISCUSSION THREAD

**Discussion by maddanio** [ansible/ansible-language-server#353] [General]
### Problem

I would like to use this lsp in sublime. I think I would really only need to be able to start it standalone, then sublimes lsp package could connect to it. But I cant find out how to just start this server.

### Solution

provide a way to start this server "standalone", i.e. not as part of some other extentsion, but on its own, on some ansible project, so sublimes lsp plugin (or that of any other ecditor like emacs/vi etc) can just connect to it to use it

**Comment by mtoohey31** [comment]
I believe this is already supported. I use this server with neovim via nvim-lspconfig, the configuration necessary for that can be found [here](https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/server_configurations/ansiblels.lua).

**Comment by maddanio** [comment]
And how do you get the ansible lsp built? Couldn't find instructions for that

**Comment by mtoohey31** [comment]
As the docs section of the page I linked mentions:

`ansible-language-server` can be installed via `npm`:
```sh
npm install -g @ansible/ansible-language-server
```

If you want to build it from source, checking the scripts in the package.json is a good place to start, I haven't done it though, so you might have to ask someone more involved in the project.

**Comment by maddanio** [comment]
ah, there, at the bottom of it :). yes, ok. Guess I will leave this open though as I don't find it easy to find this out at all.
thanks for the pointer!

**Comment by mtoohey31** [comment]
Happy to help, and you’re right, documentation about standalone usage should probably be included somewhere in this repo. Do you want to change the title to something like “Include Standalone Usage Documentation”?

**Comment by ganeshrn** [comment]
@mtoohey31 @maddanio PR on document update is welcomed.

**Comment by justaskz** [comment]
@maddanio could you please share, how to use `ansible-language-server` with sublime?

**Comment by maddanio** [comment]
uff, been a while, and I am not sure I am even using it. ansible is not currently in my job tasks. but as @mtoohey31 said above and I seem to have found it should be documented at the bottom of this page: https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/server_configurations/ansiblels.lua

**Reply by justaskz** [reply]
@maddanio Thats an instruction how to install. Did you manage to make it work? Here is my sublime lsp settings:
```
    "ansible": {
      "enabled": true,
      "command": ["ansible-language-server", "--stdio"],
      "env": { "PATH": "/opt/homebrew/bin" },
      "selector": "source.ansible",
    },
```

I get this log after each ansible file change:
```
LSP: starting ['ansible-language-server', '--stdio'] in /Users/t1028/repos/ansible
LSP: ansible: Supported semantic token types: ['method', 'class', 'keyword', 'property']
LSP: ansible: Supported semantic token modifiers: ['definition']
LSP: ansible: registering capability: didChangeConfigurationProvider
LSP: ansible: registering capability: workspace.workspaceFolders
LSP: ansible: registering capability: didChangeWatchedFilesProvider
```

But there are no markings if I make obvious syntax error

**Comment by maddanio** [comment]
Hmm, no, sorry, I don't really remember anymore to be honest.
